src/hardware/ipx.cpp

```diff
@@ -592,24 +598,39 @@ static void receivePacket(uint8_t *buffer, int16_t bufSize) {
 	LOG_IPX("IPX: RX Packet loss!");
 }
 
-static void IPX_ClientLoop(void) {
+void IPX_UDPClientListener()
+{
 	int numrecv;
 	UDPpacket inPacket;
 	inPacket.data = (Uint8 *)recvBuffer;
 	inPacket.maxlen = IPXBUFFERSIZE;
 	inPacket.channel = UDPChannel;
 
-	// Its amazing how much simpler UDP is than TCP
-	numrecv = SDLNet_UDP_Recv(ipxClientSocket, &inPacket);
-	if(numrecv) receivePacket(inPacket.data, inPacket.len);
-}
+	SDLNet_SocketSet socketset = SDLNet_AllocSocketSet(1);
+	SDLNet_UDP_AddSocket(socketset, ipxClientSocket);
+
+	while (socketset != NULL && incomingPacket.connected)
+	{
+		if (!isPinging)
+		{
+			if (SDLNet_CheckSockets(socketset, SOCKET_WAIT) == 1)
+			{
+				if (SDLNet_UDP_Recv(ipxClientSocket, &inPacket))
+					receivePacket(inPacket.data, inPacket.len);
+			}
+		}
+	}
 
+	if (socketset != NULL)
+		SDLNet_FreeSocketSet(socketset);
+}
```

Comment:
Please run all the new code through clang-format as per the development guide so it's all in our standard style (the use of braces is off, but don't try to fix it manually; just use clang-format).